### PR TITLE
feat: [OPA-1326]: harness-core-ui work for new policy pipeline tab

### PR DESCRIPTION
**Pull request by kylekstewart**
### Summary

Added new hook to return some RBAC Factory data.

Passing some resource group hooks to display and fetch resource group data

#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by kylekstewart** [comment]
retrigger jest

**Comment by kylekstewart** [comment]
retrigger jest

File: src/modules/20-rbac/pages/ResourceGroupDetails/utils.tsx [hunk 1]
```diff
@@ -380,3 +380,7 @@ export const getDefaultIncludedScope = (
 export const includesCurrentScope = (scopes: ScopeSelector[], resourceScope: Scope): boolean => {
   return !!scopes.find(scope => getScopeFromDTO(scope) === resourceScope && scope.filter === 'EXCLUDING_CHILD_SCOPES')
 }
+
+export const useGetResourceTypeHandler = (resourceType: ResourceType): ResourceHandler | undefined => {
```
Comment (pankajsingh-harness) [inline src/modules/20-rbac/pages/ResourceGroupDetails/utils.tsx:384]: NIT: 
```suggestion
export const useGetResourceType = (resourceType: ResourceType): ResourceHandler | undefined => {
```

**Review by avinashmadhwani02 — APPROVED** [review]

File: src/modules/20-rbac/pages/ResourceGroupDetails/utils.tsx (hunk 1 above)
Comment (rameshx) [reply src/modules/20-rbac/pages/ResourceGroupDetails/utils.tsx:384]: this custom hook is not returning `resourceType`, it's returning its handler so the name looks fine to me

**Review by rameshx — APPROVED** [review]

**Pull request merged**